plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -145,6 +145,142 @@ func NoDiskConflict(pod *api.Pod, existingPods []*api.Pod, node string) (bool, e
 	return true, nil
 }
 
+type MaxPDVolumeCountChecker struct {
+	filter     PDVolumeFilter
+	maxVolumes uint
+	pvInfo     PersistentVolumeInfo
+	pvcInfo    PersistentVolumeClaimInfo
+}
+
+// PDVolumeFilter contains information on how to filter PD Volumes when checking PD Volume caps
+type PDVolumeFilter struct {
```

Comment:
The name of this type is confusing because it contains a filter for normal volumes and one for persistent volumes. Would it be better to call it a VolumeFilter?